include/benchmark/benchmark_api.h

```diff
@@ -609,11 +756,15 @@ class Fixture: public internal::Benchmark {
 
     virtual void Run(State& st) {
       this->SetUp(st);
+      this->InitState(st);
       this->BenchmarkCase(st);
+      this->TerminateState(st);
       this->TearDown(st);
     }
 
     virtual void SetUp(const State&) {}
+    virtual void InitState(State &) {}
```

Comment:
i need more details why these are necessary, sorry. if the above API can add counters as well as set them, why would you need to set anything up in the fixture?